### PR TITLE
feat: allow washover-time int

### PR DESCRIPTION
**Pull request by pablobd**
In this PR, I am adding warnings when superfluous params are provided to the `PowerConfig`. Superfluous params are those that have no effect in the power simulations run. It also overrides these superfluous params to the default values. In this way, if config1 and config2 differ in superfluous params, this code `config1 == config2` will eval to True.

### CODE REVIEW

File: cluster_experiments/power_config.py [hunk 1]
```diff
@@ -29,7 +31,7 @@
 )
 
 
-@dataclass
+@dataclass(frozen=True)
```
Comment (david26694) [inline cluster_experiments/power_config.py:34]: what does this do?

**Review by david26694 — APPROVED** [review]
would you add a test for the int case? or is it too dummy?

File: cluster_experiments/power_config.py (hunk 1 above)
Comment (pablobd) [reply cluster_experiments/power_config.py:34]: This makes the object immutable and hashable.... that means that we can have a set of configs or they can be the key of a dictionary. On the other hand, people can't change the arguments of the class once instantiated

**Comment by pablobd** [comment]
> would you add a test for the int case? or is it too dummy?

Yes, I can. In which test file would you add it?

**Comment by david26694** [comment]
> > would you add a test for the int case? or is it too dummy?
> 
> Yes, I can. In which test file would you add it?

I guess in https://github.com/david26694/cluster-experiments/blob/main/tests/power_analysis/test_switchback_power.py
or https://github.com/david26694/cluster-experiments/blob/main/tests/splitter/test_washover.py, but perhaps a new one  in splitter/ makes sense too

File: cluster_experiments/power_config.py [hunk 2]
```diff
@@ -122,6 +126,88 @@ class PowerConfig:
 
     seed: Optional[int] = None
 
+    def __post_init__(self):
```
Comment (david26694) [inline cluster_experiments/power_config.py:129]: Wdyt about adding this method:

```python
    def set_and_log(self, attr, value, other_attr):
        setattr(self, attr, value)
        logging.warning(
            f"{attr} = {getattr(self, attr)} has no effect with "
            f"{other_attr} = {getattr(self, other_attr)}. "
            f"Overriding {attr} to {value}."
        )
```

then doing:

```python
        if "switchback" not in self.splitter:
            if self._are_different(self.switch_frequency, None):
                self.set_and_log("switch_frequency", None, "splitter")
```

File: cluster_experiments/power_config.py [hunk 3]
```diff
@@ -122,6 +126,88 @@ class PowerConfig:
...
+            if self._are_different(self.scale, None):
+                self.scale = None
+                logging.warning(
+                    f"{self.scale = } has no effect with {self.perturbator = }."
+                    "Overriding scale to None."
+                )
+
+        if "balanced" not in self.splitter:
+            if self._are_different(self.treatments, None):
+                self.treatments = None
```
Comment (david26694) [inline cluster_experiments/power_config.py:166]: I think treatments should work with any splitter, not only balanced splitters

File: cluster_experiments/power_config.py [hunk 4]
```diff
@@ -122,6 +126,88 @@ class PowerConfig:
...
+                logging.warning(
+                    f"{self.treatments = } has no effect with {self.splitter = }."
+                    "Overriding treatments to None."
+                )
+
+        if "stratified" not in self.splitter and "ttest" not in self.analysis:
```
Comment (david26694) [inline cluster_experiments/power_config.py:172]: We can even be more precise and do:
```suggestion
        if "stratified" not in self.splitter and "paired_ttest" not in self.analysis:
```

File: cluster_experiments/power_config.py [hunk 5]
```diff
@@ -122,6 +126,88 @@ class PowerConfig:
...
+            if self._are_different(self.strata_cols, None):
+                self.strata_cols = None
+                logging.warning(
+                    f"{self.strata_cols = } has no effect with {self.splitter = }."
+                    "Overriding strata_cols to None."
+                )
+
+        # if "cupac" not in self.analysis:
```
Comment (david26694) [inline cluster_experiments/power_config.py:180]: This chould be:
```python
if cupac_model == ""
```

**Review by david26694 — CHANGES_REQUESTED** [review]
Some minor changes, it also needs to increase version in setup. At some point I'd add the tests about the `_post_init_`, maybe we can do in a future PR

File: cluster_experiments/power_config.py (hunk 3 above)
Comment (pablobd) [reply cluster_experiments/power_config.py:166]: So, removing this block entirely!

File: cluster_experiments/power_config.py (hunk 4 above)
Comment (pablobd) [reply cluster_experiments/power_config.py:172]: thanks, I changed it

File: cluster_experiments/power_config.py (hunk 5 above)
Comment (pablobd) [reply cluster_experiments/power_config.py:180]: thanks. It's done now

File: cluster_experiments/power_config.py (hunk 2 above)
Comment (pablobd) [reply cluster_experiments/power_config.py:129]: Looks nice! I'll give it a try

File: cluster_experiments/power_config.py [hunk 6]
```diff
@@ -122,6 +126,48 @@ class PowerConfig:
 
     seed: Optional[int] = None
 
+    def __post_init__(self):
+        if "switchback" not in self.splitter:
+            if self._are_different(self.switch_frequency, None):
+                self._set_and_log("swithc_frequency", None, "splitter")
```
Comment (david26694) [inline cluster_experiments/power_config.py:132]:
```suggestion
                self._set_and_log("switch_frequency", None, "splitter")
```

**Review by david26694 — CHANGES_REQUESTED** [review]
one last change, can you increase setup version?

File: cluster_experiments/power_config.py (hunk 6 above)
Comment (pablobd) [reply cluster_experiments/power_config.py]: I got this and other minor errors thanks to the new tests added  🚀

**Review by david26694 — APPROVED** [review]
lgtm! let's see if it's released :)

**Pull request merged**